CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1

```diff
@@ -130,9 +134,22 @@ Describe "Single compute node protocol tests with utils" {
     }
 
     AfterEach {
-        Write-Host "Removing containers"
-        if (Get-Variable Container1ID -ErrorAction SilentlyContinue) {
+        function Test-IfVariableAndContainerExist {
+            Param ([Parameter(Mandatory=$true)] $ContainerID)
+
+            return $(
+                $(Get-Variable $ContainerID -ErrorAction SilentlyContinue) -and
+                $(docker ps -aq | Select-String $(Get-Variable $ContainerID))
+            )
```

Comment:
All these `$()` subexpressions could be changed to just `()` parentheses. Also, the outer `()` is not necessary.


> ```powershell
> Select-String $(Get-Variable $ContainerID)
> ```

From what I've tested, this won't work, as `Get-Variable` returns an object. Try either

```powershell
Select-String $ContainerID
```

(`-and` is lazily evaluated, so missing variable on rhs won't cause strict mode to scream) or

```powershell
Select-String (Get-Variable ContainerID).Value
```